homeassistant/components/flume/sensor.py

```diff
@@ -81,18 +82,19 @@ async def async_setup_entry(
 
     flume_auth = flume_domain_data[FLUME_AUTH]
     http_session = flume_domain_data[FLUME_HTTP_SESSION]
-    flume_devices = flume_domain_data[FLUME_DEVICES]
+    flume_devices = [
```

Comment:
Let's may this a function so we don't have duplicate code

Reply:
We don't exactly have duplicate code:

## `binary_sensor`

```python
  flume_devices = [
        device
        for device in flume_domain_data[FLUME_DEVICES].device_list
        if KEY_DEVICE_LOCATION_NAME in device[KEY_DEVICE_LOCATION]
    ]
```

## `sensor`
```python 
    flume_devices = [
        device
        for device in flume_domain_data[FLUME_DEVICES].device_list
        if KEY_DEVICE_LOCATION_NAME in device[KEY_DEVICE_LOCATION]
        and device[KEY_DEVICE_TYPE] == FLUME_TYPE_SENSOR
    ]
```

I guess I can make a function that allows or returning only sensors...

Reply:
I think you could do something like this so you don't have to teach new function about the sensor platform

https://github.com/home-assistant/core/commit/5f8ca932c8731fd4ac7745d5503f45a5fde7b67f

Reply:
Done! :)

Reply:
Did you forget to push?